internal/dim-2d/dim-2d_test.go

```diff
@@ -0,0 +1,118 @@
+
+	dim2d "github.com/viamrobotics/viam-cartographer/internal/dim-2d"
+	"github.com/viamrobotics/viam-cartographer/internal/testhelper"
+)
+
+func TestNewLidar(t *testing.T) {
+	logger := golog.NewTestLogger(t)
+
+	t.Run("No sensor provided", func(t *testing.T) {
+		sensors := []string{}
+		deps := testhelper.SetupDeps(sensors)
+		actualLidar, err := dim2d.NewLidar(context.Background(), deps, sensors, logger)
+		expectedLidar := lidar.Lidar{}
+		test.That(t, actualLidar, test.ShouldResemble, expectedLidar)
+		test.That(t, err, test.ShouldBeNil)
+	})
+
+	t.Run("Failed lidar creation due to more than one sensor provided", func(t *testing.T) {
+		sensors := []string{"lidar", "one-too-many"}
+		deps := testhelper.SetupDeps(sensors)
+		actualLidar, err := dim2d.NewLidar(context.Background(), deps, sensors, logger)
+		expectedLidar := lidar.Lidar{}
+		test.That(t, actualLidar, test.ShouldResemble, expectedLidar)
+		test.That(t, err, test.ShouldBeError,
+			errors.New("configuring lidar camera error: 'sensors' must contain only one lidar camera, but is 'sensors: [lidar, one-too-many]'"))
+	})
+
+	t.Run("Failed lidar creation with non-existing sensor", func(t *testing.T) {
+		sensors := []string{"gibberish"}
+		deps := testhelper.SetupDeps(sensors)
+		actualLidar, err := dim2d.NewLidar(context.Background(), deps, sensors, logger)
+		expectedLidar := lidar.Lidar{}
+		test.That(t, actualLidar, test.ShouldResemble, expectedLidar)
+		test.That(t, err, test.ShouldBeError,
+			errors.New("configuring lidar camera error: error getting lidar camera "+
+				"gibberish for slam service: \"gibberish\" missing from dependencies"))
+	})
+
+	t.Run("Successful lidar creation", func(t *testing.T) {
+		sensors := []string{"good_lidar"}
+		ctx := context.Background()
+		deps := testhelper.SetupDeps(sensors)
+		actualLidar, err := dim2d.NewLidar(ctx, deps, sensors, logger)
+		test.That(t, actualLidar.Name, test.ShouldEqual, sensors[0])
+		test.That(t, err, test.ShouldBeNil)
+
+		pc, err := actualLidar.GetData(ctx)
+		test.That(t, pc, test.ShouldNotBeNil)
+		test.That(t, err, test.ShouldBeNil)
+	})
+}
+
+func TestGetAndSaveData(t *testing.T) {
+	ctx := context.Background()
+	logger := golog.NewTestLogger(t)
+	dataDir, err := slamTesthelper.CreateTempFolderArchitecture(logger)
+	test.That(t, err, test.ShouldBeNil)
+
+	t.Run("Successful call to GetAndSaveData", func(t *testing.T) {
+		sensors := []string{"good_lidar"}
+		deps := testhelper.SetupDeps(sensors)
+		actualLidar, err := dim2d.NewLidar(ctx, deps, sensors, logger)
+		test.That(t, actualLidar.Name, test.ShouldEqual, sensors[0])
+		test.That(t, err, test.ShouldBeNil)
+
+		_, err = dim2d.GetAndSaveData(ctx, dataDir, actualLidar, logger)
+		test.That(t, err, test.ShouldBeNil)
+
+		files, err := os.ReadDir(dataDir + "/data/")
+		test.That(t, len(files), test.ShouldEqual, 1)
+		test.That(t, err, test.ShouldBeNil)
+	})
+
+	testhelper.ClearDirectory(t, dataDir)
+}
+
+func TestValidateGetAndSaveData(t *testing.T) {
```

Comment:
[minor] Can we add a test to check output if a bad lidar is given where say NextPointCloud returns an error as well as any other types of malformed lidars that were created back in out rdk unit tests?